tools/fuzzer.cpp

```diff
@@ -0,0 +1,403 @@
+      utf8CharBytes.emplace_back(
+          randWithinRange(limits::utf8CommonLowerBound, limits::utf8CommonUpperBound, randomGenerator));
+      break;
+    }
+    case 3: {
+      utf8CharBytes.emplace_back(randWithinRange(limits::utf83BytesFirstLowerBound, limits::utf83BytesFirstUpperBound,
+                                                 randomGenerator));
+      for (uint32_t i = 0; i < 2; ++i) {
+        utf8CharBytes.emplace_back(
+            randWithinRange(limits::utf8CommonLowerBound, limits::utf8CommonUpperBound, randomGenerator));
+      }
+      break;
+    }
+    case 4: {
+      utf8CharBytes.emplace_back(randWithinRange(limits::utf84BytesFirstLowerBound, limits::utf84BytesFirstUpperBound,
+                                                 randomGenerator));
+      for (uint32_t i = 0; i < 3; ++i) {
+        utf8CharBytes.emplace_back(
+            randWithinRange(limits::utf8CommonLowerBound, limits::utf8CommonUpperBound, randomGenerator));
+      }
+      break;
+    }
+    default:
+      assert(false);
+  }
+  utf8CharBytes.emplace_back('\0');
+  utf8Str.append(std::string(utf8CharBytes.data()));
+}
+
+static void generateUtf8String(RandomGenerator& randomGenerator, std::string& utf8Str) {
+  using limits = KnownLimitValues;
+  std::uniform_int_distribution<> distr(limits::minUtf8RandStringLength, limits::maxUtf8RandStringLength);
+  uint32_t length = distr(randomGenerator.mt64);
+  for (uint32_t i = 0; i < length; ++i) {
+    appendRandUtf8Char(randomGenerator, utf8Str);
+  }
+}
+
+template <typename Format>
+static void generateVelocypack(Builder& builder, uint32_t depth, RandomGenerator& randomGenerator) {
+  using limits = KnownLimitValues;
+  RandomBuilderAdditions maxValue = RandomBuilderAdditions::ADD_DOUBLE;
+
+  if constexpr (std::is_same_v<Format, VPackFormat>) {
+    maxValue = RandomBuilderAdditions::ADD_MAX_VPACK_VALUE;
+  }
+
+  while (true) {
+    RandomBuilderAdditions randomBuilderAdds = static_cast<RandomBuilderAdditions>(randomGenerator.mt64() %
+                                                                                   maxValue);
+    if (depth > limits::maxDepth && randomBuilderAdds <= ADD_OBJECT) {
+      continue;
+    }
+    switch (randomBuilderAdds) {
+      case ADD_ARRAY: {
+        builder.openArray(randomGenerator.mt64() % 2 ? true : false);
+        uint32_t numMembers = randomGenerator.mt64() % 10;
+        for (uint32_t i = 0; i < numMembers; ++i) {
+          generateVelocypack<Format>(builder, depth + 1, randomGenerator);
+        }
+        builder.close();
+        break;
+      }
+      case ADD_OBJECT: {
+        builder.openObject(randomGenerator.mt64() % 2 ? true : false);
+        uint32_t numMembers = randomGenerator.mt64() % 10;
+        for (uint32_t i = 0; i < numMembers; ++i) {
+          std::string key;
+          generateUtf8String(randomGenerator, key);
+          builder.add(Value(key + std::to_string(i)));
```

Comment:
This doesn't make sure that we generate unique attribute names.
For example, if we generate the string `"1"` in iteration 0, we get `"1"` + `"1"` = `"11"`.
If in iteration 11, we generate the empty string `""`, we get `""` + `"11"` = `"11"`, which is a conflict.